src/crypto/srtp.rs

```diff
@@ -0,0 +1,251 @@
+use std::fmt;
+
+use self::aead_aes_128_gcm::AeadKey;
+use self::aes_128_cm_sha1_80::AesKey;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum SrtpProfile {
+    #[cfg(feature = "_internal_test_exports")]
+    PassThrough,
+    Aes128CmSha1_80,
+    AeadAes128Gcm,
+}
+
+impl SrtpProfile {
+    // All the profiles we support, ordered from most preferred to least.
+    pub(crate) const ALL: &'static [SrtpProfile] =
+        &[SrtpProfile::AeadAes128Gcm, SrtpProfile::Aes128CmSha1_80];
+
+    /// The length of keying material to extract from the DTLS session in bytes.
+    #[rustfmt::skip]
+    pub(crate) fn keying_material_len(&self) -> usize {
+        match self {
+            #[cfg(feature = "_internal_test_exports")]
+            SrtpProfile::PassThrough => 0,
+             // MASTER_KEY_LEN * 2 + MASTER_SALT * 2
+             // TODO: This is a duplication of info that is held in srtp.rs, because we
+             // don't want a dependency in that direction.
+            SrtpProfile::Aes128CmSha1_80 => 16 * 2 + 14 * 2,
+            SrtpProfile::AeadAes128Gcm   => 16 * 2 + 12 * 2,
+        }
+    }
+}
+
+#[allow(unused)]
+
+pub fn new_aes_128_cm_sha1_80(
+    key: AesKey,
+    encrypt: bool,
```

Comment:
This would be nicer as an enum to avoid the "boolean trap":

```rust
enum Mode {
	Encrypt,
	Decrypt
}
```

TBH, I am not sure it adds much value. Personally, I'd prefer non-branching functions, even if it adds a few LoC.

Reply:
I did it like this because I felt there were already too many `#[cfg(feature = "openssl")]` so from `rtp/srtp.rs` the contract is currently 3 functions (instead of 5) which ideally I'd reduce even further.

I don't quite see a good pattern here though.